### PR TITLE
support for a validation pre-commit hooks

### PR DESCRIPTION
**Pull request by foolioo**
Pre-commit hooks for [`dasel validate`](https://github.com/TomWright/dasel/issues/219).

## Notes

1. The version specified in the documentation (v1.25.1) refers to a new release that includes `.pre-commit-hooks.yaml`
2. ~The version specified in `.pre-commit-config.yaml` (master) will work only after merging this change. For testing (see below) the commit hash for this pull-request should be used (e.g. https://github.com/TomWright/dasel/pull/232/commits/1217e03c614c14d0c98b0a7cdc5c65e93132b378). This file is optional, and needed only if [pre-commit](https://github.com/pre-commit/pre-commit) is not in used for maintaining dasel project~
3. ~I couldn't find the source code for the documentation files to update them accordingly, and that is the reason that the link in the Features section is left empty. If you will point me to them, I would love to make a contribution~

## Tests

Execute [pre-commit](https://github.com/pre-commit/pre-commit) within the repository (added for sanity tests)

```shell
$ pre-commit run -a dasel-validate --verbose
Validate JSON, YAML, XML, TOML...........................................Failed
- hook id: dasel-validate
- duration: 0.04s
- exit code: 1

fail tests/assets/broken.json could not load input: could not unmarshal data: invalid character '}' after array element
pass benchmark/data.json
pass benchmark/data.yaml
pass .github/workflows/codeql-analysis.yml
Error: 1 files failed validation
pass tests/assets/example.yaml
pass .github/FUNDING.yml
pass .github/workflows/build-test.yaml
pass benchmark/data/nested_property.json
pass benchmark/data/root_object.json
pass benchmark/data/delete_property.json
pass .github/workflows/build.yaml
pass tests/assets/deployment.yaml
pass tests/assets/example.json
pass .github/workflows/build-dev.yaml
pass benchmark/data/array_index.json
pass benchmark/data/list_array_keys.json
pass .github/workflows/bump-homebrew.yaml
pass benchmark/data/top_level_property.json
pass .pre-commit-hooks.yaml
pass tests/assets/example.xml
fail tests/assets/broken.xml could not load input: could not unmarshal data: xml.Decoder.Token() - XML syntax error on line 1: element <a> closed by </b>
pass benchmark/data/append_array_of_strings.json
pass .github/workflows/test.yaml
pass codecov.yaml
Error: 1 files failed validation
pass benchmark/data/overwrite_object.json
pass benchmark/data/update_string.json
```

cc @TomWright

### CODE REVIEW

**Comment by TomWright** [comment]
Hey @foolioo, the repo containing the docs are here: https://github.com/TomWright/dasel-docs

**Comment by TomWright** [comment]
@foolioo Yes please that would be great!

File: README.md [hunk 1]
```diff
@@ -124,6 +141,7 @@ Please [open a discussion](https://github.com/TomWright/dasel/discussions) if:
 - [Run via Docker](https://daseldocs.tomwright.me/installation#docker).
 - [Faster than jq/yq](#benchmarks).
 - [Self update](https://daseldocs.tomwright.me/installation/update).
+- [Pre-commit hooks]()
```
Comment (TomWright) [inline README.md:144]: Link to docs here please

**Review by TomWright — CHANGES_REQUESTED** [review]
> Thank you for the docs reference. Do you think the changes for the README in this change should be also added to the docs repository?

Yes please.

On another note, will this change add a pre-commit hook to the repo? This isn't ideal as it contains invalid JSON on purpose and will block commits

**Review by TomWright — APPROVED** [review]
Looks good to me. All OK for this to be merged now?

**Comment by TomWright** [comment]
Could you update the `CHANGELOG.md` file please? You'll need to add the version at the top of the file aswell as links towards the bottom.

Once that's done I'll merge and release

**Comment by aalmiray** [comment]
Hello there! I'm the author of JReleaser 😄 

Though I'd love to see more people using the tool I could not fail to notice this particular project is built with Go, in which case I strongly recommend you to use GoReleaser instead, as it provides deeper integration with Go's build toolchain https://goreleaser.com/

**Comment by TomWright** [comment]
Thanks for the suggestion @aalmiray, I'll take a look 😄

**Pull request merged**